### PR TITLE
Remove Statsd library and Statsd integration from Enterprise Search

### PR DESCRIPTION
**Pull request by rinatkhaziev**
## Description

With Prom implementation we don't need Statsd anymore.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [ ] This change works and has been tested locally (or has an appropriate fallback).
- [ ] This change works and has been tested on a Go sandbox.
- [ ] This change has relevant unit tests (if applicable).
- [ ] This change uses a rollout method to ease with deployment (if applicable - especially for large scale actions that require writes).
- [ ] This change has relevant documentation additions / updates (if applicable).
- [ ] I've created a changelog description that aligns with the provided examples.

### CODE REVIEW

File: search/includes/classes/class-queue.php [hunk 1]
```diff
@@ -1066,7 +1063,7 @@ public function ratelimit_indexing( $bail, $sync_manager, $indexable_slug ) {
 
 		// If indexing operation ratelimiting is hit, queue index operations
 		if ( $index_count_in_period > self::$max_indexing_op_count || self::is_indexing_ratelimited() ) {
-			$this->record_ratelimited_stat( $increment, $indexable_slug );
+			// TODO: swap with Prom
```
Comment (rinatkhaziev) [inline search/includes/classes/class-queue.php]: @sjinks could you please assist with that?

**Comment by rinatkhaziev** [comment]
Ready for review and more thorough testing.

File: search/includes/classes/class-search.php [hunk 2]
```diff
@@ -1109,13 +1099,13 @@ public function is_url_query_cacheable( string $url, $args ): bool {
...
 	 * @param array $query
-	 * @param string $statsd_prefix
+	 * @param string $unused unused
 	 * @param string $type
 	 * @param mixed $cached_request
 	 * @param string $query_method
 	 * @return void
 	 */
-	public function ep_handle_failed_request( $request, $response, $query, $statsd_prefix, $type, $cached_request, $query_method ) {
+	public function ep_handle_failed_request( $request, $response, $query, $unused, $type, $cached_request, $query_method ) {
```
Comment (rebeccahum) [inline search/includes/classes/class-search.php]: Will we be using `$unused` for anything in the future where we should keep it in the function signature?

Comment (rinatkhaziev) [reply search/includes/classes/class-search.php]: I doubt we would but updating method on a public method is a bit risky.

Comment (rebeccahum) [reply search/includes/classes/class-search.php]: It's not being used anywhere. Also, can you detail what type of testing is needed to be done on this removal?

Comment (rinatkhaziev) [reply search/includes/classes/class-search.php]: Addressed

**Review by rebeccahum — APPROVED** [review]

**Pull request merged**